SCORER.md

```diff
@@ -21,11 +28,278 @@ You may need to build your own scorer - your own _language model_ if:
+
+```
+~/deepspeech-data$ ls | grep indonesian-sentences
+ 476 -rw-rw-r--  1 root root  483481 Feb 24 19:02 indonesian-sentences.txt
+```
+
+The `indonesian-sentences.txt` file is formatted with one phrase per line, eg:
+
+```
+Kamar adik laki-laki saya lebih sempit daripada kamar saya.
+Ayah akan membunuhku.
+Ini pulpen.
+Akira pandai bermain tenis.
+Dia keluar dari ruangan tanpa mengatakan sepatah kata pun.
+Besok Anda akan bertemu dengan siapa.
+Aku mengerti maksudmu.
+Tolong lepas jasmu.
+```
+
+#### Using `lm_optimizer.py` to generate values for the parameters `--default_alpha` and `--default_beta` that are used by the `generate_scorer_package` script
+
+The `lm_optimizer.py` script is located in the `DeepSpeech` directory if you have set up your [environment][ENVIRONMENT.md] as outlined in the PlayBook.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# ls | grep lm_optimizer.py
+lm_optimizer.py
+```
+
+This script takes a set of test data (`--test_files`), and a `--checkpoint_dir` parameter and determines the optimal `--default_alpha` and `--default_beta` values.
+
+Call `lm_optimizer.py` and pass it the `--test_files` and a `--checkpoint_dir` directory.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# python3 lm_optimizer.py \
+     --test_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/test.csv \
+     --checkpoint_dir deepspeech-data/checkpoints
+```
+
+Remember, if you trained your model with a `--n_hidden` value that is different to the default (`1024`), you should pass the same `--n_hidden` value to `lm_optimizer.py`, for example:
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# python3 lm_optimizer.py \
+     --test_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/test.csv \
+     --checkpoint_dir deepspeech-data/checkpoints \
+     --n_hidden 4
+```
+
+`lm_optimizer.py` will create a new _study_.
+
+```
+[I 2021-03-05 02:04:23,041] A new study created in memory with name: no-name-38c8e8cb-0cc2-4f53-af0e-7a7bd3bc5159
+```
+
+It will then run _testing_ and output a trial score.
+
+```
+[I 2021-03-02 12:48:15,336] Trial 0 finished with value: 1.0 and parameters: {'lm_alpha': 1.0381777700987271, 'lm_beta': 0.02094605391055826}. Best is trial 0 with value: 1.0.
+```
+
+By default, `lm_optimizer.py` will run `6` trials, and identify the trial with the most optimal parameters.
+
+```
+[I 2021-03-02 17:50:00,662] Trial 6 finished with value: 1.0 and parameters: {'lm_alpha': 3.1660260368070423, 'lm_beta': 4.7438794403688735}. Best is trial 0 with value: 1.0.
+```
+
+The optimal parameters `--default_alpha` and `--default_beta` are now known, and can be used with `generate_scorer_package`. In this case, the optimal settings are:
+
+```
+--default_alpha 1.0381777700987271
+--default_beta 0.02094605391055826
+```
+
+because `Trial 0` was the best trial.
+
+##### Additional parameters for `lm_optimizer.py`
+
+There are additional parameters that may be useful.
```

Comment:
Maybe warn that this is an expensive step, and its not unlikely to take a few days with some parameters ? hardware

Reply:
Good call